src/tr2/game/objects/general/grenade.c

```diff
@@ -119,17 +119,7 @@ void __cdecl Grenade_Control(int16_t item_num)
         } else {
             // XXX: missing check if obj is intelligent?
             Gun_HitTarget(target_item, NULL, 30);
-
             explode = true;
-            g_SaveGame.statistics.hits++;
-
-            if (target_item->hit_points <= 0) {
-                g_SaveGame.statistics.kills++;
```

Comment:
I think this is the only line that needs removed. Currently, grenade hits aren't recorded and the enemy doesn't actually explode, just dies normally.

Reply:
Oops yes. I misread the TR2Main code.

Reply:
I'm confused though. Because `Gun_HitTarget` should be checking:
```
    if (item->hit_points > 0 && item->hit_points <= damage) {
        g_SaveGame.statistics.kills++;
    }
```